lib/fs.js

```diff
@@ -1579,6 +1579,17 @@ function watch(filename, options, listener) {
   if (listener) {
     watcher.addListener('change', listener);
   }
+  if (options.signal) {
+    if (options.signal.aborted) {
+      process.nextTick(() => watcher.close());
```

Comment:
Why on next tick?

Reply:
So that users have a chance to subscribe to the `"close"` event.

Reply:
(Just because `watch` returns a watcher and not a promise or a callback)